include/sparrow/config/config.hpp

```diff
@@ -38,7 +38,37 @@
 #   define SPARROW_API __attribute__((visibility("default")))
 #endif
 
-consteval bool is_apple_compiler()
+namespace sparrow
 {
-    return static_cast<bool>(COMPILING_WITH_APPLE_CLANG);
+    consteval bool is_apple_compiler()
+    {
+        return static_cast<bool>(COMPILING_WITH_APPLE_CLANG);
+    }
+
+    namespace config
+    {
+        inline constexpr bool enable_size_limit_runtime_check =
```

Comment:
this is for testing / debugging purpose, it is disabled by default in Release.

Reply:
> If a user is using 32 bit its, its in their responsibility to not overflow

We provide this so that users that use 64bit systems but want the data to be limited to 32bit can check that it is in fact the case and not discover the issues too late. And for testing purpose indeed. It is the same logic as allowing the user to enable assertions on their side.